go/private/binary.bzl

```diff
@@ -33,11 +35,32 @@ def _go_binary_impl(ctx):
     gc_linkopts=gc_linkopts(ctx),
     x_defs=ctx.attr.x_defs)
 
-  return struct(
-      files = depset([ctx.outputs.executable]),
-      runfiles = lib_result.runfiles,
-      cgo_object = lib_result.cgo_object,
-  )
+  # Static linking (in the 'static' output group)
+  static_linkopts = [
```

Comment:
If we are going to add lots of variations on this theme, we might have to come up with a nice way to do this repeatedly, otherwise this function is going to get very repetitive and fragile.